src/mesh/mesh.cpp

```diff
@@ -1200,17 +1200,19 @@ void Mesh::Initialize(int res_flag, ParameterInput *pin) {
         auto &pbval = pmb->pbval;
         if (multilevel) pbval->ProlongateBoundaries(0.0, 0.0);
 
-        int il = pmb->is, iu = pmb->ie, jl = pmb->js, ju = pmb->je, kl = pmb->ks,
-            ku = pmb->ke;
-        if (pbval->nblevel[1][1][0] != -1) il -= NGHOST;
-        if (pbval->nblevel[1][1][2] != -1) iu += NGHOST;
+        IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::interior);
+        IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::interior);
+        IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::interior);
+
+        if (pbval->nblevel[1][1][0] != -1) ib.s -= NGHOST;
+        if (pbval->nblevel[1][1][2] != -1) ib.e += NGHOST;
         if (pmb->block_size.nx2 > 1) {
-          if (pbval->nblevel[1][0][1] != -1) jl -= NGHOST;
-          if (pbval->nblevel[1][2][1] != -1) ju += NGHOST;
+          if (pbval->nblevel[1][0][1] != -1) jb.s -= NGHOST;
+          if (pbval->nblevel[1][2][1] != -1) jb.e += NGHOST;
         }
         if (pmb->block_size.nx3 > 1) {
-          if (pbval->nblevel[0][1][1] != -1) kl -= NGHOST;
-          if (pbval->nblevel[2][1][1] != -1) ku += NGHOST;
+          if (pbval->nblevel[0][1][1] != -1) kb.s -= NGHOST;
+          if (pbval->nblevel[2][1][1] != -1) kb.e += NGHOST;
         }
```

Comment:
Is this even being used? we are creating il, iu, etc but there are never sent anywhere?

Reply:
Hmm this is weird. I wonder if these loops originally set the global `is,ie,js,je,ks,ke` variables owned by each mesh block.  They might have been made local in a refactor. 

This makes me worried that the global variables might not be getting set correctly somewhere.

Reply:
Thanks for the prompt reply.

Reply:
I guess the thing to do is to ensure the regression tests pass with your WIP MR, now that you wrote them. :) In particular, the AMR advection.

Reply:
Should I just delete the dead code? It is likely that it was these particular indices were defined locally somewhere in stead and someone simply forgot to remove this part.

Reply:
Normally, I'd say yes. This one makes me a little nervous though. Can we comment it out for now?

Reply:
I think it is best to delete it, but I'll see if someone else knows why it is hanging around. @jdolence might know whether it is safe to remove this chunk, he touched code nearby. 

These indices were originally used with:

```
pmb->peos->ConservedToPrimitiveCellAverage(ph->u, ph->w1, pf->b,
                                                     ph->w, pf->bcc, pmb->pcoord,
                                                     il, iu, jl, ju, kl, ku);
if (NSCALARS > 0) {
   pmb->peos->PassiveScalarConservedToPrimitiveCellAverage(
   ps->s, ps->r, ps->r, pmb->pcoord, il, iu, jl, ju, kl, ku);
}
```

Reply:
> These indices were originally used with:

Ah, if they were called in `con2prim` and passive scalar stuff, then we can safely delete them.  That code will live elsewhere in the new code, I think.

Reply:
As Jonah noted, in Athena++ these indices are passed to ConservedToPrimitive just below this code.  It sounds like this call has been deleted.  Originally it was there (I think) because prolongation uses the primitive variables.  We need to preserve this functionality in some way, although obviously not with this exact code per se.

Reply:
Perhaps it's worth putting in a comment here with a TODO note, to ensure we keep track of this as a point in the code where we may wish to extract primitives?